sql/core/src/test/scala/org/apache/spark/sql/SparkSessionExtensionSuite.scala

```diff
@@ -171,6 +172,30 @@ class SparkSessionExtensionSuite extends SparkFunSuite {
     }
   }
 
+  test("reset column vectors") {
+    val session = SparkSession.builder()
+      .master("local[1]")
+      .config(COLUMN_BATCH_SIZE.key, 2)
+      .withExtensions { extensions =>
+        extensions.injectColumnar(session =>
+          MyColumarRule(PreRuleReplaceAddWithBrokenVersion(), MyPostRule())) }
+      .getOrCreate()
+
+    try {
+      assert(session.sessionState.columnarRules.contains(
+        MyColumarRule(PreRuleReplaceAddWithBrokenVersion(), MyPostRule())))
+      import session.sqlContext.implicits._
+
+      val input = Seq((100L), (200L), (300L))
+      val data = input.toDF("vals").repartition(1)
+      val df = data.selectExpr("vals + 1")
+      val result = df.collect()
+      assert(result sameElements input.map(x => Row(x + 2)))
+    } finally {
+      stop(session)
+    }
+  }
```

Comment:
BTW, `SparkSessionExtensionSuite.scala` seems to be an integration test for

- sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala

Could you make a UT at one of the following packages, please?

- sql/core/src/test/scala/org/apache/spark/sql/execution/
- sql/core/src/test/scala/org/apache/spark/sql/execution/columnar

Reply:
Hi @dongjoon-hyun , thanks for your comments.
I put this UT here because the spark session is built with extensions which inject the columnar rules and it rely on the existing code in the same file. In fact the most part of this UT reused the code in `test("inject columnar")`.
Should I just move this UT under the directory you mentioned ? I haven't find a proper file to place it yet. Is it good if I create a new test suite file? e.g.

- sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/InjectColumnarSuite.scala

Reply:
Ur, no~ Not moving this test case. What I meant was a small new UT inside some test suites in that packages.